posts/tools/test-frame-introduction.md

```diff
@@ -0,0 +1,174 @@
+And, one day, you will find yourself copying the testing-related classes across multiple repositories,
+mixing things and always trying to refactor the code without success and without "the light at the end of the tunnel" that
+there will be an end to this.
+At least, that is the situation in which we ended up.
+
+Yes, you can use the test classes from the main repository as a Maven dependency, in case you are releasing it together
+with other modules to the Maven Central (or anywhere else).
+However, there are situations when you cannot do that - one of the examples is cyclic dependency.
+
+Let's say that you have the main repository, containing the code for an operator, and you have a few other repositories for the components that
+are managed by the operator.
+Each of the components has a different release cadence, so having them in the operator's repository is not the correct approach.
+In this particular case, if you would use the "system test" dependency from the operator's repository inside the component's repository (and the component
+is used inside the operator's repository), then you create a cyclic dependency, where the component waits for the release of the operator
+with new things in the test classes, and the operator waits for the component to be released, so it can be included in the controller's code.
+
+Or, you can have one repository for the test logic, that will be shared across multiple repositories, and thanks to that you will remove
+the cyclic dependency.
+Nevertheless, with a new separate repository, there are different challenges.
+You need to think about how well it will be maintained - do you and your team have time to actively contribute to this repository and release a new
+version of it regularly, without breaking the logic of the test suites in each of the repositories?
+How frequent will be the releases?
+What about the situation when you work on different repositories outside of the current organization and this module cannot be imported?
+
+## Test-Frame as the solution
+
+Because Skodjob consists of engineers working on different projects, but with the same passion for testing the integration between the components,
+we were finding a proper solution for sharing the code between organizations, that would help us with removing redundancy and duplicated code, which
+was anyway present throughout the repositories.
+And because we had the same questions I mentioned in the previous section, we decided to create a library called [Test-Frame](https://github.com/skodjob/test-frame/).
+The library aims to provide basic functionality for testing applications running on Kubernetes.
+It also tries to make the overall process as easy as possible.
+
+### What it provides?
+
+The core of the Test-Frame is `KubeResourceManager` which is responsible for the management of resources during the test phases.
+For its functioning, we need to pass Fabric8-based objects, that are extending the `HasMetadata` class, to it.
```

Comment:
```suggestion
For it to function, we need to pass Fabric8-based objects that extend the `HasMetadata` class to it.
```

Reply:
`For it to function` seems weird, the current sentence makes sense. Or maybe start the sentence differently 🤷

Reply:
looks good, thanks